### ISSUE TITLE
[BUG]: Accessibility Checker fails when running in Ubuntu latest version 24.4

### ISSUE THREAD

**Issue by carlosparreno** [IBMa/equal-access#2186]
### Project

accessibility-checker for Node

### Operating system

Other

### Description

We are getting the following error in our accessibility unit tests written with `accessibility-checker` for node when running in Ubuntu 24 latest LTS Version.

<img width="418" alt="Image" src="https://github.com/user-attachments/assets/ec056c29-1658-4d89-ab56-6f9db18cd2b8" />

We get this error per unit test.

It seems that Ubuntu 24.04 has implemented [stronger restrictions](https://ubuntu.com/blog/whats-new-in-security-for-ubuntu-24-04-lts#:~:text=22.04%20LTS.-,Unprivileged%20user%20namespace%20restrictions,-Unprivileged%20user%20namespaces) around unprivileged user namespaces, not sure if this is the reason of failures.

Many possible solutions we found in the internet are around calling puppeteer with the `--no-sandox` flag, but also it seems not to be recommended by the Ubuntu troubleshooting message.

This ticket might be useful: https://github.com/puppeteer/puppeteer/issues/12818#issuecomment-2250443567

It might be an issue in puppeteer rather than in `accessibililty-checker`, not 100% sure but the version of puppeteer used by `accessibility-checker` seems quite outdated and potentially related.

### Steps to reproduce

1. Create unit tests using the `accessiblity-checker` apis. This is our code, it loops an array of components that we want to run a11y tests on:
```
describe('accessibility tests', () => {
  Object.keys(Examples).forEach(label => {
    it(`${label}`, async () => {
      const component = renderInBody(Examples[label]);
      const results = await getCompliance(component, label);
      expect(JSON.stringify(assertCompliance(results.report))).toEqual('0');
    }, 30000);
  });

  afterAll(() => close());
});
```
2. Run the accessibility test suit in UBUNTU 24.04 and the following error appears.

<img width="418" alt="Image" src="https://github.com/user-attachments/assets/8b110fb4-6d85-45b9-b2b0-5ad71a18e121" />

<img width="1585" alt="Image" src="https://github.com/user-attachments/assets/e3c50028-5770-407b-beb9-a22ccfabb6ae" />

**Comment by tombrunet** [comment]
@carlosparreno A couple of different things here...

1) Yes, the puppeteer version in 3.x is outdated. We have some 4.x release candidates that use a newer version of Puppeteer. We expect to have a release out in the coming weeks for 4.x
2) accessibility-checker has a dependency on Puppeteer for scanning based on URL input. However, I think your `renderInBody` is likely passing in a parsed element/document. In that scenario, we do not instantiate Puppeteer, but rather your code does.

So, I suggest you dig into the `renderInBody` and find where you are instantiating Puppeteer and determine what version is being used there and what configuration you are creating Puppeteer with.

**Comment by carlosparreno** [comment]
Hi @tombrunet, thanks a lot for your quick reply,

We don't use puppeteer per se, it's not installed in our project, only as a subdependency of `accessibility-checker` and other unrelated library (storybook to display our components).

As per your explanation, I don't think puppeteer should be instantiated then... This is our code for `renderInBody`, as you assumed it's just a parsed document.

```
const renderInBody = component => {
  return `
  <!DOCTYPE html>
  <html lang=en-US>
    <head>
      <title>${component}</title>
    </head>
    <body>
      <main role="main">
        ${render(component)}
      </main>
    </body>
  </html>
  `;
};
```

However, the error is there when running it on Ubuntu 24.04. 

Any ideas of why?

**Comment by tombrunet** [comment]
Okay, so you're passing in the page as a string? We'll have to look into a config parameter then.

**Comment by henriquemattos** [comment]
Hi, team.

I don't want to deviate from the second topic being discussed here, but I'm having the same problem regarding `--no-sandbox` in `ubuntu-latest` running on GitHub Actions. 

I have a simple HTML page using Bulma.io as CSS framework and FontAwesome. There's no JavaScript or other dependence.

I execute locally and it succeeds, but on GitHub Actions it fails with:

```shell
Run npm run achecker-ci
  npm run achecker-ci
  shell: /usr/bin/bash -e {0}
> niteroi@1.0.0 achecker-ci
> PUPPETEER_SKIP_CHROMIUM_DOWNLOAD=true achecker --outputFormat html --outputFolder report ./websrc/index.html
Error: Failed to launch the browser process!
[0226/204013.894081:FATAL:zygote_host_impl_linux.cc(117)] No usable sandbox! Update your kernel or see https://chromium.googlesource.com/chromium/src/+/main/docs/linux/suid_sandbox_development.md for more information on developing with the SUID sandbox. If you want to live dangerously and need an immediate workaround, you can try using --no-sandbox.
#0 0x564fcac35b89 base::debug::CollectStackTrace()
#1 0x564fcab9a433 base::debug::StackTrace::StackTrace()
#2 0x564fcabad330 logging::LogMessage::~LogMessage()
#3 0x564fc8b9299b content::ZygoteHostImpl::Init()
#4 0x564fca743372 content::ContentMainRunnerImpl::Initialize()
#5 0x564fca741439 content::RunContentProcess()
#6 0x564fca74158e content::ContentMain()
#7 0x564fca79cd7a headless::(anonymous namespace)::RunContentMain()
#8 0x564fca79ca85 headless::HeadlessShellMain()
#9 0x564fc72003e8 ChromeMain
#10 0x7f333c62a1ca (/usr/lib/x86_64-linux-gnu/libc.so.6+0x2a1c9)
#11 0x7f333c62a28b __libc_start_main
#12 0x564fc720022a _start
TROUBLESHOOTING: https://github.com/puppeteer/puppeteer/blob/main/docs/troubleshooting.md
    at onClose (/home/runner/work/niteroi/niteroi/node_modules/puppeteer/lib/cjs/puppeteer/node/BrowserRunner.js:241:20)
    at Interface.<anonymous> (/home/runner/work/niteroi/niteroi/node_modules/puppeteer/lib/cjs/puppeteer/node/BrowserRunner.js:231:68)
    at Interface.emit (node:events:530:35)
    at Interface.close (node:internal/readline/interface:526:10)
    at Socket.onend (node:internal/readline/interface:252:10)
    at Socket.emit (node:events:530:35)
    at endReadableNT (node:internal/streams/readable:1698:12)
    at process.processTicksAndRejections (node:internal/process/task_queues:90:21)
node:internal/process/promises:394
    triggerUncaughtException(err, true /* fromPromise */);
    ^
``` 

My command is in package.json is `achecker --outputFormat html --outputFolder report ./websrc/index.html`.

DevDependence is `"accessibility-checker": "^4.0.1",`

Any idea how to run this on GitHub Actions without having to instantiate a whole JavaScript test framework behind/around?

Kind regards,
Henrique

**Comment by tombrunet** [comment]
Addressed in this PR: https://github.com/IBMa/equal-access/pull/2251
Will show up in the next release.

**Comment by philljenkins** [comment]
Re-open this issue if not fixed in the next release.

**Comment by tombrunet** [comment]
@henriquemattos This should be deployed now. See the referenced PR for parameters you would probably add to the config for Ubuntu.

**Comment by henriquemattos** [comment]
Great, I'm currently focused on another task, but I guess up to the weekend I will have time to try it. I can see that version 4.0.4 is already on NPM Registry, so should be easy to consume. 

Thank you very much, @tombrunet .
